### PR TITLE
tests/test_suite_pem: Augment DES test cases with AES: PEM

### PR DESCRIPTION
**Pull request by tgonzalezorlandoarm**
## Description

A few negative test cases in test_suite_pem.data rely on DES (“invalid iv”, “malformed”). DES is deprecated.
Construct similar test cases using AES and use them to remove the DES ones.

This is in response to issue #7037 

## PR checklist

- [x] **changelog**: Not needed because this is a change in the test data.
- [x] **backport** : https://github.com/Mbed-TLS/mbedtls/pull/7901
- [x] **tests**: Provided

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

### CODE REVIEW

File: tests/suites/test_suite_pem.data
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_pem.data:25]: Please keep the DES test cases.

And I think we should backport. We generally backport test coverage improvements if they're relevant and not too hard.

Comment (gilles-peskine-arm) [reply tests/suites/test_suite_pem.data]: We usually don't create the backport until the main PR has two approvals.

Comment (gilles-peskine-arm) [reply tests/suites/test_suite_pem.data]: Please squash the commits before review. We generally keep commits as they come, but [remove x] followed by [restore x] is an unusual case where it's best to squash.

The review includes the commit history, so we'd have to re-review after squashing anyway. As much as possible, we avoid rewriting the history once review has started.

Comment (gilles-peskine-arm) [reply tests/suites/test_suite_pem.data]: Ah, ok, I'd understood “there are no more suggested changes” as after review. Please go on then.

**Comment by tgonzalezorlandoarm** [comment]
All tests are passing in commit 9aec5351063ab67da7ec5ef7139f47f64cbb8646
Will rebase now

**Review by yanesca — APPROVED** [review]
LGTM.

**Comment by waleed-elmelegy-arm** [comment]
LGTM.

Note: I don't know if this is part of the contributing standard or if it makes any difference but usually PRs are made from a feature branch from your forked development branch not from the development branch directly

**Review by waleed-elmelegy-arm — APPROVED** [review]
LGTM.

Note: I don't know if this is part of the contributing standard or if it makes any difference but usually PRs are made from a feature branch from your forked development branch not from the development branch directly, maybe someone else can confirm.

**Comment by yanesca** [comment]
> Note: I don't know if this is part of the contributing standard or if it makes any difference but usually PRs are made from a feature branch from your forked development branch not from the development branch directly, maybe someone else can confirm.

Yep, it is a better practice like that, as that way you don't need to force-push your own development branch all the time and can have several PRs at the same time.

**Comment by gilles-peskine-arm** [comment]
I in fact recommend not having a local branch called `development`, because it gets confusing when you forget to fast-forward it. Instead, if you want `development` to be meaningful as a branch name, use 
```
git symbolic-ref development refs/remotes/upstream-public/development
```
(adjust for your remote name) so that the name `development` will always be up to date after a `git fetch`.

**Comment by tgonzalezorlandoarm** [comment]
#7901 is the backport.

**Pull request merged**
